File: app/src/main/java/com/chess/cryptobot/content/history/HistoryPreferences.java
package com.chess.cryptobot.content.history;

import android.content.Context;

import com.chess.cryptobot.content.Preferences;

public class HistoryPreferences extends Preferences {
    HistoryPreferences(Context context) {
        super(context);
    }

    @Override
    protected String initPrefKey(Context context) {
        return "history";
    }

    @Override
    public void addItem(String itemName) {
    }
}

File: app/src/main/java/com/chess/cryptobot/content/history/HistoryPreferences.kt
package com.chess.cryptobot.content.history

import android.content.Context
import com.chess.cryptobot.R

import com.chess.cryptobot.content.Preferences

class HistoryPreferences internal constructor(context: Context?) : Preferences(context) {

    override fun initPrefKey(context: Context?): String {
        return context?.resources?.getString(R.string.history_pref_key) ?: "history"
    }

    override fun addItem(itemName: String) {}
}
